Initial definition of targets for the Sail C and OCaml emulators.

pmundkur opened the pull request:
The Sail formal model generates two emulators, C and OCaml.  This adds a target for each of them.  Both are passing all current tests.

pmundkur commented:
Hi, could someone give me some preliminary feedback on this pull request?

eroom1966 commented:
I have reviewed these updates, and they do not affect anything else in the repository, self contained within riscv-target, so I think this is OK to merge

I think the question is, how does this get tested ?
If I try
make RISCV_TARGET=sail-riscv-c
>> Makefile.include:3: *** Target simulator executable 'riscv_c_sim_RV32` not found. Stop.

or 
make RISCV_TARGET=sail-riscv-ocaml
>> Makefile.include:3: *** Target simulator executable 'riscv_ocaml_sim_RV32` not found. Stop.

Should we have some guidance on how to make these targets execute the code produced ?

Thx
Lee

jeremybennett commented:
Hi Prashanth,

Could you update the doc/README.adoc file with guidance, addressing the issues raised by Lee.

Could you also update the ChangeLog files accordingly (guidance here: https://www.gnu.org/prep/standards/html_node/Change-Logs.html) and incorporate the body of the ChangeLog entry in your commit message.

Thanks,


Jeremy

pmundkur commented:
Thanks for the review!

I've updated the README with instructions on using the model, which will need to be built and the PATH suitably updated in the environment.  I also updated the Changelog, and force-pushed to get a clean commit if merged.  Do the instructions now provide sufficient guidance?

pmundkur commented:
Hi @eroom1966 and @jeremybennett ,  does my revised pull request provide the information you wished about using the Sail model?

jeremybennett commented:
Sorry, but we still have conflicts (GitHub makes a mess of merging ChangeLogs). Could you fix and force-push and I'll merge.

pmundkur commented:
@jeremybennett  thanks for the review; I fixed the conflict and force-pushed.

jeremybennett commented:
All merged - thanks for your contribution.
